File: SV.Server/Controllers/Models/CardSearchRequest.cs
using System.Collections.Generic;
using SV.Server.Services.Constants;
using SV.Server.Services.Models;

namespace SV.Server.Controllers.Models
{
    public class CardSearchRequest
    {
        public CraftType? Craft { get; init; }
        public string Name { get; init; }
        public IList<RarityType> Rarities { get; init; }
        public IList<CardType> Types { get; init; }

        internal SearchCardRequest ToRequest()
        {
            return new SearchCardRequest
            {
                Craft = this.Craft,
                Name = this.Name,
                Rarities = this.Rarities,
                Types = this.Types
            };
        }
    }
}

File: SV.Web/src/app/shared/types/api/card-search-request.ts
import { CardType } from './../customs/card-type.enum';
import { Rarity } from './../customs/rarity.enum';
import { Craft } from '../customs/craft.enum';

export interface CardSearchRequest {
  craft: Craft;
  name: string;
  rarities: Rarity[];
  types: CardType[];
}
